src/Build/Logging/BaseConsoleLogger.cs

```diff
@@ -966,16 +966,21 @@ public int Compare(object o1, object o2)
                     {
                         return 0;
                     }
-                    else if (p1.reenteredScope)
+                    else if (p1.reenteredScope && !p2.reenteredScope)
                     {
                         // p1 was reentrant; sort first
                         return -1;
                     }
-                    else
+                    else if (!p1.reenteredScope && p2.reenteredScope)
                     {
                         // p2 was reentrant; sort first
                         return 1;
                     }
+                    else
+                    {
+                        // both reentrant; sort properly to avoid throwing
```

Comment:
nit: instead of "properly" say "stably by another field", so the motivation is clearer.